tla-types/src/test/scala/at/forsyte/apalache/tla/typecheck/TlaType1Gen.scala

```diff
@@ -0,0 +1,99 @@
+package at.forsyte.apalache.tla.typecheck
+
+import org.scalacheck.Arbitrary.arbitrary
+import org.scalacheck.Gen
+import org.scalacheck.Gen.{choose, const, identifier, listOfN, lzy, oneOf, resize, sized}
+
+// Generators for the case classes of TlaType1
+trait TlaType1Gen {
+  def genConst: Gen[TlaType1] =
+    for {
+      str <- identifier
+    } yield ConstT1(str.toUpperCase())
+
+  def genVar: Gen[TlaType1] =
+    for {
+      i <- arbitrary[Int]
+      // produce an absolute value. Note that Math.abs(Integer.MIN_VALUE) = Integer.MIN_VALUE, so use max(0, abs(i)).
+    } yield VarT1(Math.max(0, Math.abs(i)))
```

Comment:
Should this use `Gen.posNum`? https://github.com/typelevel/scalacheck/blob/8645105297ff26512d002e87d413717212892977/src/main/scala/org/scalacheck/Gen.scala#L1243 (Usage here: https://github.com/typelevel/scalacheck/blob/2224f8506ca8ab079e3a0c4d18d30db7a0395e70/jvm/src/test/scala/org/scalacheck/PropertyFilterSpecification.scala#L16)